pkg/kubelet/container_manager_linux.go

```diff
@@ -142,26 +142,48 @@ func createManager(containerName string) *fs.Manager {
 	}
 }
 
-const sysctlVmOvercommitMemory = "vm/overcommit_memory"
-
-// disableKernelMemoryOvercommitHandling tells the kernel to perform no memory over-commit handling.
-// Under this setting, the potential for memory overload is increased, but so is performance for
-// memory-intensive tasks
-// sets /proc/sys/vm/overcommit_memory to 1
-func disableKernelMemoryOvercommitHandling() error {
-	val, err := utilsysctl.GetSysctl(sysctlVmOvercommitMemory)
-	if err != nil {
-		return err
-	}
-	if val == 1 {
-		return nil
+// TODO: plumb this up as a flag to Kubelet in a future PR
+type KernelTunableBehavior string
```

Comment:
Essentially these kernel features are hard requirements and we should treat them that way.
Since these changes are implicit requirements for QoS, shouldn't we at-least mark the node as `NotReady`? 
From that perspective, I don't see a reason for supporting the `Warn` option.